### PR TITLE
[code fixes] Fix some issues found by LGTM

### PR DESCRIPTION
**Pull request by jotak**
See https://lgtm.com/projects/g/kiali/kiali-ui/alerts

- Secure external links with rel="noopener noreferrer"
- Remove unecessary condition checks
- Remove unecessary assignments
- Un-hide "LoginState" imported type

### CODE REVIEW

File: src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ class DestinationRuleDetail extends React.Component<DestinationRuleProps> {
     let message = checks.map(check => check.message).join(',');
 
     if (!message.length) {
-      if (validation && !validation.valid) {
+      if (!validation.valid) {
```
Comment (cfcosta) [inline src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx]: Is this really the same? Something like `!undefined.method` returns a `TypeError` on the console.

Comment (josejulio) [reply src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx]: Was wondering the same, but validation is never undefined up to this point, check line [30](https://github.com/jotak/swsui/blob/2d90dc92c79d38d7da1caeb3a52456c15f64863d/src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx#L30)

File: src/components/About/AboutUIModal.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ class AboutUIModal extends React.Component<AboutUIModalProps, AboutUIModalState>
     if (config().about && config().about.website) {
       return (
         <div>
-          <a href={config().about.website.url} target="_blank">
+          <a href={config().about.website.url} target="_blank" rel="noopener noreferrer">
```
Comment (mtho11) [inline src/components/About/AboutUIModal.tsx]: This` rel="noopener noreferrer"` is new to me. nice.

File: src/reducers/LoginState.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import { getType } from 'typesafe-actions';
-import { LoginState } from '../store/Store';
+import { LoginState as LoginStateInterface } from '../store/Store';
```
Comment (mtho11) [inline src/reducers/LoginState.ts]: I have never seen an `as` in an `import`.

File: src/components/About/AboutUIModal.tsx (hunk 2 above)
Comment (jotak) [reply src/components/About/AboutUIModal.tsx]: I learnt this yesterday too :)

File: src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx (hunk 1 above)
Comment (jotak) [reply src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/DestinationRuleDetail.tsx]: yes and normally typescript is smart enough to report an error if we potentially access a field from undefined variable (unless it's `any`)

**Pull request merged**